clippy_lints/src/matches.rs

```diff
@@ -388,6 +419,28 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Matches {
             }
         }
     }
+
+    fn check_pat(&mut self, cx: &LateContext<'a, 'tcx>, pat: &'tcx Pat<'_>) {
```

Comment:
This will lint all patterns, not only the once in `match` expressions. This itself is fine, IMO, but the lint name should reflect this.

`REST_PAT_IN_FULLY_BINDED_STRUCTS` maybe?

Reply:
I wasn't able to find reasons to not lint all patterns, but yes I agree, the lint name should be changed.

This new name sounds good to me.